File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.java
class Solution {
    public int solution(int[] arr) {
        int lcm = arr[0];

        for (int i = 1; i < arr.length; i++) {
            lcm = lcm * arr[i] / gcd(lcm, arr[i]);
        }

        return lcm;
    }

    // 최대공약수 (유클리드 호제법)
    private int gcd(int a, int b) {
        while (b != 0) {
            int temp = a % b;
            a = b;
            b = temp;
        }
        return a;
    }
}

File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.js
function solution(arr) {
    let lcm = arr[0];

    for (let i = 1; i < arr.length; i++) {
        lcm = lcm * arr[i] / gcd(lcm, arr[i]);
    }

    return lcm;
}

// 최대공약수 (유클리드 호제법)
function gcd(a, b) {
    while (b !== 0) {
        let temp = a % b;
        a = b;
        b = temp;
    }
    return a;
}
